writer.go

```diff
@@ -335,6 +338,55 @@ func NewMediaPlaylist(winsize uint, capacity uint) (*MediaPlaylist, error) {
 	return p, nil
 }
 
+// InsertSegments allows the insertion of one or multiple MediaSegments into the MediaPlaylist.
+// seqID is the sequence ID at which the new segments should be inserted to
+func (p *MediaPlaylist) InsertSegments(segments []MediaSegment, seqID uint64) error {
+	if len(segments) == 0 {
+		return ErrPlaylistEmpty
+	}
+
+	// Create copies of the segments to be inserted
+	localSegments := make([]*MediaSegment, len(segments))
+	for i, segment := range segments {
+		copySegment := segment // Create a shallow copy of the segment
+		localSegments[i] = &copySegment
+	}
+
+	adjustment := uint(len(localSegments))
+	p.count += adjustment
+	p.tail += adjustment
+
+	// Determine the index where the new segments should be inserted
+	var insertIndex int
+	// Allow passing in 0 for prerolls
```

Comment:
While this was done in the context of ads, since this is a general library, we should work to abstract comments. Best to re-phrase this along the lines of "allow for insertion at the beginning of the playlist". It does unfortunately more describe "what"  (instead of "why", as I'll usually advocate for in comments) but this could in theory be used outside modifying the manifest for ads.